etl/tests/test_flask_app.py

```diff
@@ -34,6 +36,27 @@ def test_client(monkeypatch: _pytest.monkeypatch.MonkeyPatch, database_name: str
     return flask_app.App.test_client()
 
 
+@pytest.fixture()
+def thread_runner() -> typing.Generator:
+    yield flask_app.ThreadRunner
+    while flask_app.ThreadRunner.is_thread_running():
+        time.sleep(0.1)
+
+
+def test_threadrunner(thread_runner: flask_app.ThreadRunner) -> None:
+    stay_asleep = True
+
+    def sleeper() -> None:
+        while stay_asleep:
+            pass
+
+    thread_runner.start_new_thread(sleeper)
+    assert thread_runner.is_thread_running()
+    stay_asleep = False
+    time.sleep(0.1)
```

Comment:
I think instead of sleeping, it'd be better to explicitly wait until the thread is done. This can be done by `join`ing the thread. As it is the thread is hidden away as a private variable of ThreadRunner, but to get around that we could (and I think there is value in) add a `join` method to ThreadRunner that just calls join on the internal thread.